routes/soldiers.js

```diff
@@ -0,0 +1,30 @@
+const soldiers = require('../collections/soldiers.js');
+const urlib = require('url');
+
+function handleSoldierReq(req, soldierData, done) {
+
+  const parsedUrlString = req.url.split("/")
+  if (parsedUrlString.length > 3) {
+    done(new Error())
+  }
+  const soldierId = parsedUrlString[2];
+  const parsedUrl = new urlib.URL("http://localhost:3000/" + req.url);
+  if (req.method === "GET") {
+    if (parsedUrlString.length === 3 && /tt[0-9]{7}/.test(soldierId)) {
+      soldiers.findSoldier({"id": soldierId}, done)
```

Comment:
return soldiers.find...

Reply:
Add this for every line of code which calls the callback or sends the callback as a callback for another function